File: 2021/03/Part1.cpp
#include <string>
#include <vector>

#include "lib.h"

int main() {
    std::vector<std::string> lines = readLines("input.txt");

    std::string gamma = "";
    std::string epsilon = "";
    for (int i = 0; i < lines.at(0).length(); i++) {
        int num_zeros = 0;
        int num_ones = 0;
        for (std::string line: lines) {
            if (line.at(i) == '0') {
                num_zeros += 1;
            } else {
                num_ones += 1;
            }
        }
        if (num_zeros > num_ones) {
            gamma += "0";
            epsilon += "1";
        } else {
            gamma += "1";
            epsilon += "0";
        }
    }

    int g = strtoull(gamma.c_str(), nullptr, 2);
    int e = strtoull(epsilon.c_str(), nullptr, 2);
    printf("%d\n", g * e);

    return 0;
}

File: 2022/01/Part1.py
elves = []
current_elf = 0
with open("input.txt") as f:
    for line in f:
        line = line[:-1]
        if line == "":
            elves.append(current_elf)
            current_elf = 0
        else:
            current_elf += int(line)

print(max(elves))